eclipse-scout-core/src/table/columns/LookupEditor.ts

```diff
@@ -0,0 +1,192 @@
+      tree.on('nodesInserted', e => {
+        tree.setNodeExpandedRecursive(e.nodes, true);
+        tree.invalidateLayoutTree();
+      });
+    }
+
+    // Workaround for unexpected reset of popup size when lookup structure resizes
+    // TODO [25.1]: Remove when a better solution has been implemented (#379394)
+    this.session.layoutValidator.schedulePostValidateFunction(() => {
+      const layout = this._popup.htmlComp.layout as PopupLayout;
+      layout.autoSize = false;
+      this._popup.position(true);
+    });
+
+    lookupBox.one('lookupCallDone', e => this._popup.open());
+
+    this._popup.on('close', e => {
+      this._popup = null;
+      cellEditorPopup.completeEdit();
+    });
+
+    lookupBox.refreshLookup();
+  }
+
+  protected _createLookupBoxModel(): ChildModelOf<LookupBox<TValue>> {
+    if (this.browseHierarchy) {
+      return {
+        objectType: TreeBox<TValue>,
+        lookupCall: this.lookupCall,
+        value: this.value,
+        cssClass: 'no-mandatory-indicator',
+        labelVisible: false,
+        statusVisible: false,
+        tree: {
+          objectType: Tree,
+          checkable: true,
+          autoCheckChildren: true
+        }
+      };
+    }
+    return {
+      objectType: ListBox<TValue>,
+      lookupCall: this.lookupCall,
+      value: this.value,
+      cssClass: 'no-mandatory-indicator',
+      labelVisible: false,
+      statusVisible: false,
+      table: {
+        objectType: Table,
+        parent: this,
+        checkable: true,
+        checkableStyle: Table.CheckableStyle.CHECKBOX_TABLE_ROW,
+        headerVisible: false,
+        footerVisible: false,
+        columns: [{
+          objectType: Column,
+          autoOptimizeWidth: true,
+          autoOptimizeMaxWidth: 450
+        }]
+      }
+    };
+  }
+}
+
+export interface LookupEditorModel<TValue> extends ValueFieldModel<TValue[]> {
+  lookupCall?: LookupCallOrModel<TValue>;
+  /**
+   * Default is false.
+   */
+  browseHierarchy?: boolean;
+  /**
+   * A positive number, _not_ null or undefined!
+   *
+   * Default is SmartField.DEFAULT_BROWSE_MAX_COUNT.
+   */
+  browseMaxRowCount?: number;
+}
+
+export interface LookupEditorLookupCallDoneEvent<TValue = any, TSource extends LookupEditor<TValue> = LookupEditor<TValue>> extends Event<TSource> {
+  result: LookupResult<TValue>;
+}
+
+export interface LookupEditorPrepareLookupCallEvent<TValue = any, TSource extends LookupEditor<TValue> = LookupEditor<TValue>> extends Event<TSource> {
+  lookupCall: LookupCall<TValue>;
+}
+
+export interface LookupEditorEventMap<TValue> extends ValueFieldEventMap<TValue[]> {
+  'lookupCallDone': LookupEditorLookupCallDoneEvent<TValue>;
+  'prepareLookupCall': LookupEditorPrepareLookupCallEvent<TValue>;
+}
+
+export class LookupEditorTreeLayout extends TreeLayout {
+
+  constructor(tree: Tree) {
+    super(tree);
+  }
+
+  override preferredLayoutSize($container: JQuery, options?: HtmlCompPrefSizeOptions): Dimension {
```

Comment:
Explain that we need this to get the preferred width of the tree nodes. A fixed height limits the view range which would only render the first few nodes.